src/app/(drops)/projects/[project]/drops/[drop]/edit/(steps)/schedule/page.tsx

```diff
@@ -12,23 +12,52 @@ import {
 } from '../../../../../../../../../providers/DropFormProvider';
 import { useDropForm } from '../../../../../../../../../hooks/useDropForm';
 import { useProject } from '../../../../../../../../../hooks/useProject';
+import { combineDateTime } from '../../../../../../../../../modules/time';
 
 export default function EditDropTimingPage() {
   const router = useRouter();
   const { project } = useProject();
   const store = useDropForm() as StoreApi<DropFormState>;
 
   const timing = useStore(store, (store) => store.timing);
+
   const setTiming = useStore(store, (store) => store.setTiming);
 
-  const { handleSubmit, register, watch, formState } = useForm<TimingSettings>({
+  const { handleSubmit, register, watch, formState, setError } = useForm<TimingSettings>({
     defaultValues: timing,
   });
 
   const selectStartDate = watch('selectStartDate');
   const selectEndDate = watch('selectEndDate');
 
   const submit = (data: TimingSettings) => {
+    let startDateTime: Date | null = null;
+    if (data.selectStartDate === 'specifyStartDate' && data.startTime && data.startDate) {
+      const [startTimeHrs, startTimeMins] = data.startTime.split(':');
+      startDateTime = combineDateTime(
+        new Date(data.startDate),
+        parseInt(startTimeHrs),
+        parseInt(startTimeMins)
+      );
+    }
+
+    let endDateTime: Date | null;
+    if (data.selectEndDate === 'specifyEndDate' && data.endDate && data.endTime) {
+      const [endTimeHrs, endTimeMins] = data.endTime.split(':');
+      endDateTime = combineDateTime(
+        new Date(data.endDate),
+        parseInt(endTimeHrs),
+        parseInt(endTimeMins)
+      );
+
+      if (startDateTime && endDateTime < startDateTime) {
+        setError('selectEndDate', {
+          message: 'End date/time cannot be before start date/time.',
+        });
+        return;
+      }
+    }
```

Comment:
The validation should be done through the validate function of the field.

https://react-hook-form.com/api/useform/register/

Reply:
Oh sorry I see you used the validate for them. You can access the current values from within the validate function so you can do this check. The second argument is the current values